apis/python/src/tiledbsoma/_sparse_nd_array.py

```diff
@@ -269,9 +270,17 @@ def write(
             Experimental.
         """
 
-        tiledb_create_options = TileDBCreateOptions.from_platform_config(
-            platform_config
-        )
+        write_options: Union[TileDBCreateOptions, TileDBWriteOptions]
+        sort_coords = None
+        if isinstance(platform_config, TileDBCreateOptions):
+            warnings.warn(
+                "The write parameter now takes in TileDBWriteOptions instead of TileDBCreateOptions",
+                DeprecationWarning,
+            )
+            write_options = TileDBCreateOptions.from_platform_config(platform_config)
+        else:
+            write_options = TileDBWriteOptions.from_platform_config(platform_config)
+            sort_coords = write_options.sort_coords
```

Comment:
@johnkerl Need your feedback here. The `write` function has been taking in `TileDBCreateOptions` so that the user can set the `consolidate_and_vacuum` flag. This parameter is only used for writes so would be appropriate to also move it into `TileDBWriteOptions`. I've added this `DeprecationWarning` here for now so that it can still take in `TileDBCreateOptions` without breaking pre-exsting code while also allowing the new `TileDBWriteOptions`. I am not sure if this is what I am supposed to do though.

Reply:
Yes @nguyenv that's perfect, and having the consolidate-and-vacuum in create options always was an oversight.

Please fill a tracking issue in GH reminding us to remove consolidate-and-vacuum from create options in, say, tiledbsoma 1.13.

Reply:
#2579